File: voom-fe/src/app/shared/rest/driver/driver-activity.api.ts
import {Api} from '../api';
import {ApiClient} from '../api-client';
import {DriverStateChangeDto} from './driver-activity.model';
import {RequestConfig} from '../rest.model';

export class DriverActivityApi extends Api {

  constructor(apiClient: ApiClient) {
    super(apiClient);
  }

  getDriverState(userId: number) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.get<void, DriverStateChangeDto>(`/api/activity/${userId}`, config);
  }

  public changeDriverState(dto: DriverStateChangeDto) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.post<DriverStateChangeDto, DriverStateChangeDto>(`/api/activity`, dto, config);
  }

}

File: voom-app/app/src/main/java/inc/visor/voom/app/shared/api/DriverActivityApi.java
package inc.visor.voom.app.shared.api;

import inc.visor.voom.app.shared.dto.driver.DriverStateChangeDto;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface DriverActivityApi {

    @POST("/api/activity")
    Call<DriverStateChangeDto> changeDriverState(@Body DriverStateChangeDto dto);

    @GET("/api/activity/{userId}")
    Call<DriverStateChangeDto> getDriverState(@Path("userId") long userId);
}